plaidml/edsl/tests/edsl_test.cc

```diff
@@ -1658,42 +1658,194 @@ TEST_F(CppEdsl, Tan) {
   checkClose(program, {input}, {expected}, /*tolerance=*/1e-4);
 }
 
-TEST_F(CppEdsl, Scatter1D) {
+TEST_F(CppEdsl, Scatter1DSum) {
+  auto D = Placeholder(DType::FLOAT32, {8});
   auto I = Placeholder(DType::INT32, {4});
   auto U = Placeholder(DType::FLOAT32, {4});
-  auto S = Placeholder(DType::INT32, {8});
-  auto O = scatter(U, I, S);
-  auto program = makeProgram("scatter", {I, U, S}, {O});
+  auto O = scatter(D, I, U);
+  auto program = makeProgram("scatter", {D, I, U}, {O});
 
+  // Don't bother initialize 'data' in SUM mode.
+  // Only the shape of data is needed.
+  std::vector<float> data;
   std::vector<int32_t> indices = {4, 3, 1, 7};
```

Comment:
Is this testing indices with same value? shouldn't indices contain same values like {4,3,3,7}?

Reply:
Yes. Will add more test cases with duplicate indices later.